zio-http/src/main/scala/zhttp/service/server/content/handler/ServerResponseHandler.scala

```diff
@@ -0,0 +1,113 @@
+package zhttp.service.server.content.handler
+
+import io.netty.buffer.{ByteBuf, Unpooled}
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.{ChannelHandlerContext, SimpleChannelInboundHandler}
+import io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR
+import io.netty.handler.codec.http.HttpVersion.HTTP_1_1
+import io.netty.handler.codec.http._
+import zhttp.core.Util
+import zhttp.http.{HTTP_CHARSET, HttpData, Response, Status}
+import zhttp.service.server.ServerTimeGenerator
+import zhttp.service.{ChannelFuture, HttpRuntime}
+import zio.stream.ZStream
+import zio.{UIO, ZIO}
+
+/**
+ * Transforms an internal Response into a HttpResponse.
+ */
+@Sharable
+private[zhttp] case class ServerResponseHandler[R](runtime: HttpRuntime[R], serverTime: ServerTimeGenerator)
+    extends SimpleChannelInboundHandler[Either[Throwable, Response[R, Throwable]]](false) {
+  override def channelRead0(ctx: ChannelHandlerContext, response: Either[Throwable, Response[R, Throwable]]): Unit = {
```

Comment:
```suggestion
  override def channelRead0(ctx: ChannelHandlerContext, response: Response[R, Throwable]): Unit = {
```

Let's handle any kind of Response over here.